pallets/nfts/src/tests.rs

```diff
@@ -1847,16 +1877,32 @@ fn burn_works() {
 			account(5),
 			default_item_config()
 		));
-		assert_eq!(Balances::reserved_balance(account(1)), 2);
 
+		// Throws error `Error::NoItemOwned` if the account does not have an entry in
+		// `AccountBalance`.
+		let account_balance = AccountBalance::take(0, account(5));
+		assert_noop!(
+			Nfts::burn(RuntimeOrigin::signed(account(5)), 0, 42),
+			Error::<Test>::NoItemOwned
+		);
+		AccountBalance::set(0, account(5), account_balance);
```

Comment:
It is good that you are testing the error case of logic that we have added yet this error can never happen as `NoPermission` is protecting against it. If you want to add it I would prefer to see a separate test. This also clearly demonstrates to the reviewer that the added logic is tested.